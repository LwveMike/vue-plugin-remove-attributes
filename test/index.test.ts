import { describe, expect, it } from 'vitest'
import { compileTemplate } from '@vue/compiler-sfc'
import type { NodeTransform } from '@vue/compiler-core'
import createAttributesRemover from '../src'

describe('createAttributesRemover', () => {
  const template
    = `<template>
      <div>
        <header data-test="header-test">
          Header
        </header>
        <section :data-test="'section-content'">
          <ul>
            <li
              :data-test="'item-text'"
              v-text="'text'"
            />
          </ul>
          <div data-meta="not-secure-meta">
            Meta 
          </div>
        </section>
        <footer dataTest="footer-test">
          Footer
        </footer>
      </div>
    </template>`

  const parse = (removeParameters: NodeTransform) => {
    return compileTemplate({
      id: 'TestComponent.vue',
      filename: 'TestComponent.vue',
      source: template,
      compilerOptions: {
        nodeTransforms: [removeParameters],
      },
    }).code
  }

  describe('empty attributes', () => {
    it('should not change anything', () => {
      const code = parse(createAttributesRemover([]))

      expect(code).toMatchSnapshot()
    })
  })
  describe('with attributes', () => {
    it('should remove all data-test and data-meta', () => {
      const code = parse(createAttributesRemover(['data-test', 'data-meta']))

      expect(code).toMatchSnapshot()
    })
  })
})
