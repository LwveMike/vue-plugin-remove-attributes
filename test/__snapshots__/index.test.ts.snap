// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createAttributesRemover > empty attributes > should not change anything 1`] = `
"import { createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = /*#__PURE__*/_createElementVNode(\\"div\\", null, [
  /*#__PURE__*/_createElementVNode(\\"header\\", { \\"data-test\\": \\"header-test\\" }, \\" Header \\"),
  /*#__PURE__*/_createElementVNode(\\"section\\", { \\"data-test\\": 'section-content' }, [
    /*#__PURE__*/_createElementVNode(\\"ul\\", null, [
      /*#__PURE__*/_createElementVNode(\\"li\\", {
        \\"data-test\\": 'item-text',
        textContent: 'text'
      })
    ]),
    /*#__PURE__*/_createElementVNode(\\"div\\", { \\"data-meta\\": \\"not-secure-meta\\" }, \\" Meta \\")
  ]),
  /*#__PURE__*/_createElementVNode(\\"footer\\", { dataTest: \\"footer-test\\" }, \\" Footer \\")
], -1 /* HOISTED */)
const _hoisted_2 = [
  _hoisted_1
]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"template\\", null, _hoisted_2))
}"
`;

exports[`createAttributesRemover > with attributes > should remove all data-test and data-meta 1`] = `
"import { createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = /*#__PURE__*/_createElementVNode(\\"div\\", null, [
  /*#__PURE__*/_createElementVNode(\\"header\\", null, \\" Header \\"),
  /*#__PURE__*/_createElementVNode(\\"section\\", null, [
    /*#__PURE__*/_createElementVNode(\\"ul\\", null, [
      /*#__PURE__*/_createElementVNode(\\"li\\", { textContent: 'text' })
    ]),
    /*#__PURE__*/_createElementVNode(\\"div\\", null, \\" Meta \\")
  ]),
  /*#__PURE__*/_createElementVNode(\\"footer\\", { dataTest: \\"footer-test\\" }, \\" Footer \\")
], -1 /* HOISTED */)
const _hoisted_2 = [
  _hoisted_1
]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"template\\", null, _hoisted_2))
}"
`;
